[Bugfix] Shifter gas/blades not set after shifting

MmmCrits opened the pull request:
Closes #21

**Notes for the reviewer:**
Fixed blades/gas not being restored after shifting back to human.

**Contributor guidelines:**
1. If applicable, a build has been uploaded to #builds to be reviewed by our testers
2. Assure the code is inline with the [AoTTG2 coding conventions](https://github.com/AoTTG-2/AoTTG-2/wiki/Code-&-Style-Conventions)

After all above conditions are met, @Lithrun or @rc174945 will review your Pull Request. Resolve any comments of the reviewer or discuss them if you disagree.

MmmCrits committed:
fixed issue 21

Fixed blades/gas not being restored after shifting back to human.

rc174945 commented:
Has this been tested? I'm not sure you can reference an old destroyed Human because it becomes null.

MmmCrits commented:
Not rigorously, but it seems to work. 
My understanding is that although the gameobject is destroyed, since the Human object is still being referenced by previousHuman then the garbage collector/whatever is managing the memory doesn't delete the Human object until nothing else references it.

rc174945 commented:
looks good, but can you make a few changes:
- It's not good practice to reference destroyed components like Human, instead lets hold a reference to the Weapon object since that's not a monobehavior
- Avoid using the 'this' keyword
- Public variables like previous should be PascalCase
- You dont need to write two cases for ThunderspearWeapon and GunWeapon, instead just use AmmoWeapon to take care of both
- Instead of using 'as' just use normal casting

MmmCrits committed:
fixed bad code

MmmCrits commented:
Thanks, I fixed those. I don't have formal programming experience, so I appreciate you for taking the time to tell me these things.

rc174945 commented on Assembly/Scripts/Characters/Human/Human.cs:
for bladeweapon and ammoweapon, if we're going to be casting 3 or more times lets declare a new variable so we only cast once. Casting is bad for performance if we do it too much

rc174945 commented on Assembly/Scripts/Characters/Shifters/BaseShifter.cs:
```diff
@@ -74,6 +76,7 @@ protected IEnumerator WaitAndBecomeHuman(float time)
             StartCoroutine(WaitAndDie());
             yield return new WaitForSeconds(2f);
             _inGameManager.SpawnPlayerAt(false, BaseTitanCache.Neck.position);
+            StartCoroutine(((Human)_inGameManager.CurrentCharacter).WaitAndTransformFromShifter(PreviousHumanGas, PreviousHumanWeapon));
```
use CurrentCharacter.StartCoroutine instead of doing StartCoroutine. The coroutine should belong to the human not the game manager

rc174945 commented on Assembly/Scripts/Characters/Human/Human.cs:
```diff
@@ -274,9 +274,37 @@ public void SpecialActionState(float time)
+            {
+                ((BladeWeapon)Weapon).BladesLeft = ((BladeWeapon)previousHumanWeapon).BladesLeft;
+                ((BladeWeapon)Weapon).CurrentDurability = ((BladeWeapon)previousHumanWeapon).CurrentDurability;
+                if (((BladeWeapon)Weapon).CurrentDurability == 0)
```
for bladeweapon and ammoweapon, if we're going to be casting 3 or more times lets declare a new variable so we only cast once. Casting is bad for performance if we do it too much

rc174945 reviewed (changes requested):
a couple more comments

MmmCrits committed:
issue 21 fix 2

Fixed bad code practices,
Fixed visual bug with broken blades / empty thunder spears after shifting.
